File: ecole/002/ecole002.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const numbers = input[1].split(' ').map(Number);

const prefixSum = new Array(n + 1).fill(0);
for (let i = 0; i < n; i++) {
    prefixSum[i + 1] = prefixSum[i] + numbers[i];
}

let result = '';
for (let k = 2; k < m + 2; k++) {
    const [i, j] = input[k].split(' ').map(Number);
    result += (prefixSum[j] - prefixSum[i - 1]) + '\n';
}
console.log(result.trim());

File: WEEK1/002/ecole002.py
import sys

n, m = map(int, sys.stdin.readline().split())
numbers = list(map(int, sys.stdin.readline().split()))

prefix_sum = [0] * (n + 1)
for i in range(n):
    prefix_sum[i + 1] = prefix_sum[i] + numbers[i]

for _ in range(m):
    i, j = map(int, sys.stdin.readline().split())
    result = prefix_sum[j] - prefix_sum[i - 1]
    print(result)
